packages/discord.js/src/client/websocket/WebSocketShard.js

```diff
@@ -835,23 +835,27 @@ class WebSocketShard extends EventEmitter {
           );
           this.connection.terminate();
         }
+
         // Emit the destroyed event if needed
-        if (emit) this._emitDestroyed();
+        if (emit) {
+          this._emitDestroyed();
+        } else if (
+          this.connection?.readyState === WebSocket.CLOSING ||
+          this.connection?.readyState === WebSocket.CLOSED
```

Comment:
```suggestion
          this.connection.readyState === WebSocket.CLOSING ||
          this.connection.readyState === WebSocket.CLOSED
```
We don't need optional chaining because connection object is never null in that scope.